Source/WebCore/Modules/mediasource/ManagedMediaSource.cpp

```diff
@@ -63,5 +70,85 @@ bool ManagedMediaSource::isTypeSupported(ScriptExecutionContext& context, const
     return MediaSource::isTypeSupported(context, type);
 }
 
+void ManagedMediaSource::startStreaming()
+{
+    if (m_streaming)
+        return;
+    m_streaming = true;
+    scheduleEvent(eventNames().startstreamingEvent);
```

Comment:
There is a small risk of quickly enqueuing 3 events, say startstreaming/endstreaming/startstreaming.
I guess the threshold should be high enough it does not happen in practice and we can always revisit this if needed.

Reply:
If the player deleted buffered data as soon as `endstreaming` was fired, I don't see how we could avoid this.
And is it a problem?